packages/core/saved-objects/core-saved-objects-migration-server-internal/src/core/unused_types.ts

```diff
@@ -49,6 +49,8 @@ export const REMOVED_TYPES: string[] = [
   'upgrade-assistant-telemetry',
   // Removed in 8.8 https://github.com/elastic/kibana/pull/155204
   'endpoint:user-artifact',
+  // Removed in 8.15.0, replaced by 'server-counters' and 'ui-counters'
+  'usage-counters',
```

Comment:
Food for thought, I wonder what happens in serverless, when a new version is started that removes the `'usage-counters'` saved objects, whilst the old version adds new ones.

We will lose partial counting, and leave some garbage until next update, but other than that it should be safe I believe.

Reply:
Since these are for telemetry, it's fine to momentarily lose some counts.

Reply:
Just FMI: does this imply a reindex? Are they removed? Or they're just hanging there forever?

Reply:
@jloleysens added a check that is rightfully failing:
```
Removing mapped properties is disallowed >8.8. Found the following properties in current mappings but not in next mappings:
[
  "usage-counters.properties.domainId"
]
```

I was removing the original usage-counters type and relocating it, thus its schema is now part of a different index (`.kibana_usage_counters`).

If we don’t keep it… the migration will not be able to update the mappings (cannot remove fields), it’ll be considered a breaking migration => reindex, which is not allowed in serverless.

I will update the PR.